### DISCUSSION TITLE
Plot tidak muncul di Shiny

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#581] [Q&A] [answered]
Saya memiliki kendala ketika menjalankan shiny. Ada plot yang tidak muncul.

Kode di UI  : 
<img width="478" alt="image" src="https://user-images.githubusercontent.com/106010284/221835572-0380aac3-985b-4b72-84f3-731d68677b12.png">

Kode di server :  
<img width="674" alt="image" src="https://user-images.githubusercontent.com/106010284/221837205-b1c08aec-75be-444a-8c2f-48055bb7948c.png">

Saat dijalankan, tidak ada error yang muncul. 

Mohon bantuannya kak, terima kasih.

**Comment by idoali** [comment]
Terima kasih atas pertanyaannya pak

Alasan mengapa tidak bisa adalah karena fungsi yang bapak gunakan, `graph_from_data_frame` tidak bisa ditampilkan pada shiny dengan fungsi `renderPlot`. Karena fungsi `graph_from_data_frame` adalah fungsi dari library `igraph`. 

Bapak bisa coba untuk mengganti fungsi `plot()` pada file server menjadi `plot.igraph()`. 

Semisal plot yang bapak mau buat bisa dibuat dengan ggplot, kami sarankan untuk membuatnya dengan ggplot saja pak.

**Comment by yohanespm77** [comment]
Sudah saya coba seperti ini mas pakai plot.igraph()
<img width="642" alt="image" src="https://user-images.githubusercontent.com/106010284/222054298-dc83d180-d286-4ac1-9beb-275a475cc821.png">

Tetapi tetap tidak muncul gambarnya.


Pakai ggplot juga tidak bisa
<img width="684" alt="image" src="https://user-images.githubusercontent.com/106010284/222054444-ec43664d-5221-41fa-822c-e4abe948d0b8.png">

**Comment by idoali** [comment, marked as answer]
Terima kasih atas tanggapannya pak

Penggunaan `ggplot` yang kami maksud adalah membuat plot dengan ggplot seperti biasa pak. Jadi inputnya bukan `graph_from_data_frame`. 

Apakah pembuatan plot dengan `graph_from_data_frame` diatas sudah dicoba di RMarkdown pak?

**Reply by yohanespm77** [reply]
Di rmarkdown sudah dicoba mas, outputnya seperti ini :
<img width="644" alt="image" src="https://user-images.githubusercontent.com/106010284/222116050-6d46b856-624b-4971-bc73-b45da1cc1e44.png">

Kalau melalui ggplotnya inputnya seperti apa jadinya mas ?

**Comment by VictorNugraha** [comment]
Dear @yohanespm77 ,

Apakah memungkinkan untuk mengshare keseluruhan filenya, agar kami dapat mencobanya juga.

Terima kasih.

Best Regards,
Team Algoritma

**Comment by yohanespm77** [comment]
Dear mas @VictorNugraha 

Berikut filenya mas : 
[shiny network.zip](https://github.com/teamalgoritma/community/files/10881950/shiny.network.zip)

**Reply by VictorNugraha** [reply]
Baik Pak, izin saya coba terlebih dahulu untuk filenya.

**Reply by yohanespm77** [reply]
Izin mas @VictorNugraha , apakah sudah ada solusinya mas ?

**Reply by VictorNugraha** [reply]
Dear Pak @yohanespm77 ,

Untuk hal tersebut kami pun belom berhasil untuk menyelesaikannya. Akan kami coba kabari jika sudah berhasil Pak.

Terima kasih
